src/components/RegisterPage.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import CreateOrLoginComponent from "./UI/CreateOrLoginComponent";
+import Form from "./UI/Form";
+import Input from "./UI/Input";
+import Button from "./UI/Button";
+import { useInput } from "../hooks/useInput.hook";
+import { AuthRequest } from "../models/request/AuthRequest";
+import { useRegisterUserMutation } from "../services/AuthService";
+
+const Container = styled.div`
+  margin-top: 5rem;
+  max-width: 20rem;
+  width: 100%;
+  height: auto;
+  margin: 7rem auto;
+`;
+
+const RegisterPage = () => {
+  const email = useInput({
+    initialValue: "",
+    validations: { isEmail: true, isEmpty: true },
+  });
+  const password = useInput({
+    initialValue: "",
+    validations: { minLength: 4, maxLength: 32, isEmpty: true },
```

Comment:
just fyi: you can use html5 validator eg
https://www.w3schools.com/tags/att_input_maxlength.asp